src/flags.js

```diff
@@ -825,4 +827,18 @@ async function mergeUsernameEmailChanges(history, targetUid, uids) {
 	}, []));
 }
 
+async function parseUsernameFilters(filters) {
+	const usernameFilters = ['assignee', 'reporterId', 'targetUid'];
+	const filtersClone = { ...filters };
+
+	await Promise.all(usernameFilters.map(async (filterKey) => {
+		if (!filtersClone[filterKey] || Number.isInteger(Number(filtersClone[filterKey]))) {
```

Comment:
One concern is that this would fail to match a username that consists of just numbers.